apps/speed.c

```diff
@@ -561,52 +639,514 @@ static OPT_PAIR ecdh_choices[] = {
 };
 #endif
 
+#ifndef SIGALRM
+# define COND(d) (count < (d))
+# define COUNT(d) (d)
+#else
+# define COND(c) (run && count<0x7fffffff)
+# define COUNT(d) (count)
+#endif                         /* SIGALRM */
+
+static unsigned char *buf = NULL, *buf2 = NULL;
+static int j;
+static char *engine_id = NULL;
+
+
+#ifndef OPENSSL_NO_MD2
+static unsigned char md2[MD2_DIGEST_LENGTH];
```

Comment:
The main reason for this is that all the variables are initialized in the main function first and then are used inside the jobs, i.e. inside the `_loop()` functions.

An alternative would have been to pass these variables using `array_loopargs` but when we wrote this implementation we realized that the readability of the code was really bad.
Also, we would have had to include all the arguments for all the operations in that struct, which would have become unnecessarily big.

Eventually we thought that the global variables were a better solution in this case since speed is self-contained in one file and these variables are not used elsewhere.